pytgcalls/pytgcalls/mtproto/telethon_bridge.py

```diff
@@ -137,21 +137,27 @@ async def edit_group_call_member(self, peer, volume: int = None, muted=False):
         self.client._handle_update(response)
 
     async def get_and_set_self_peer(self):
-        self.my_peer = await self.client.get_me()
+        self.my_peer = await self.client.get_me(input_peer=True)
 
         return self.my_peer
 
     async def get_and_set_group_call(self, group):
+
         self.chat_peer = group
-        # TODO add if cond
-        self.chat_peer = await self.client.get_input_entity(group)
-        # TODO add more if cond
-        self.full_chat = (await self.client(functions.channels.GetFullChannelRequest(group))).full_chat
+
+        if type(group) not in [InputPeerChannel, InputPeerChat]:
+            self.chat_peer = await self.client.get_input_entity(group)
+
+        if isinstance(self.chat_peer, InputPeerChannel):
+            self.full_chat = (await self.client(functions.channels.GetFullChannelRequest(group))).full_chat
+        elif isinstance(self.chat_peer, InputPeerChat):
+            self.full_chat = (await self.client(functions.messages.GetFullChatRequest(group))).full_chat
 
         if self.full_chat is None:
             raise RuntimeError(f'Can\'t get full chat by {group}')
 
         self.group_call = self.full_chat.call
+        self.join_as = self.full_chat.groupcall_default_join_as
```

Comment:
this is doesnt work...

Reply:
cuz you are resetting join_as field in resolve_and_set_join_as method... wrong place to handle default join as

Reply:
Hmm. Let me test. But maybe the default needs to be first set with `SaveDefaultGroupCallJoinAsRequest`

Reply:
I think calling the SaveDefaultGroupCallJoinAsRequest method its not a part of this lib

Reply:
Works in telethon @MarshalX

Reply:
@MarshalX  It works just fine on telethon. On pyrogram it might have to do with how caching is implemented and how the library does requests.


Telegram sets the user default join as the connected user unless is set manually using `SaveDefaultGroupCallJoinAsRequest ` to another entity.
The following code:

```
from telethon import TelegramClient

from pytgcalls import GroupCallFactory

from telethon.tl import functions


import asyncio

client = TelegramClient("Test", 111111, "1111ccc")
async def _main():
    await client.start()
    result = await client(functions.channels.GetFullChannelRequest(
        channel='@tgcallschat'
    ))
    print(result.full_chat)


asyncio.get_event_loop().run_until_complete(_main())
```

Returns this as an object:

```
ChannelFull(id=1412793637, about='GitHub: https://github.com/MarshalX/tgcalls \n\nDocumentation: https://tgcalls.org/\n\nOur awesome list: https://github.com/tgcalls/awesome-tgcalls', read_inbox_max_id=11544, read_outbox_max_id=11544, unread_count=0, chat_photo=Photo(id=5361653717549561307, access_hash=8521575912258100621, file_reference=b'\x00`\xf2\xafp\xba\x9eS\n\x1d\xfc2\x9eVT\x8d\xb4p`\x19J', date=datetime.datetime(2021, 3, 14, 1, 35, 44, tzinfo=datetime.timezone.utc), sizes=[PhotoSize(type='a', w=160, h=160, size=5462), PhotoSize(type='b', w=320, h=320, size=13532), PhotoSize(type='c', w=640, h=640, size=34363), PhotoStrippedSize(type='i', bytes=b'\x01((\xcb\xa7"3\x9c(\xcd".\xe6\x03\xd6\xb4\x11\x02(\x00Pg9\xf2\x95>\xcb&:\n\x8d\x91\x90\xfc\xc3\x15\xaac\xda@\'\x93\xed\xc5D\xe8$R\r;\x10\xaa\xbb\xeaf\xd1J\xcaU\x8a\x9e\xa2\x8aF\xe4\x96\xdf\xeb\xd75\xa4\x01\xc1\xc7Z\xc9F(\xc1\x87j\xd3\x8d\xd6E\x04\x1aL\xc2\xacu\xb9!\xc0=]\xb0i\x8cI\x19=sO\x0f\xb4\x92\x01\xc9\xed\x9e*\t\x1cF\xa5\x98\xd3L\xc7\xc9\x14\xee\x7f\xd7\xb5\x15\x1b\xb1v,{\xd1A\xd8\xb4V\x12\x9c\x8e\xc8r\xa7\x14Q@\xd9\'\xda\xa5\xc7QQ33\x1c\xb1$\xd1E\x02I-\x84\xa2\x8a(\x19')], dc_id=2, has_stickers=False, video_sizes=[VideoSize(type='u', w=480, h=480, size=708710, video_start_ts=None)]), notify_settings=PeerNotifySettings(show_previews=True, silent=None, mute_until=datetime.datetime(2038, 1, 19, 3, 14, 7, tzinfo=datetime.timezone.utc), sound=None), bot_info=[BotInfo(user_id=208056682, description='This is an official bot created by @GroupHelpBot which can help the main bot in groups.', commands=[]), BotInfo(user_id=162726413, description='This is the most complete Bot to help you manage your groups easily and safely!\n\nJust add me to a group as Admin and then use /settings to set up functions.', commands=[BotCommand(command='reload', description='Update Admins list'), BotCommand(command='ban', description='Ban a user [username/id/reply]'), BotCommand(command='kick', description='Kick a user [username/id/reply]'), BotCommand(command='warn', description='Warn a user [username/id/reply]'), BotCommand(command='unban', description='Unban a user [username/id/reply]'), BotCommand(command='info', description='User informations [username/id/reply]'), BotCommand(command='staff', description='Group Staff list'), BotCommand(command='config', description='Bot settings'), BotCommand(command='help', description='Bot commands explanation')])], pts=14608, can_view_participants=True, can_set_username=False, can_set_stickers=False, hidden_prehistory=False, can_set_location=False, has_scheduled=False, can_view_stats=False, blocked=False, participants_count=258, admins_count=None, kicked_count=None, banned_count=None, online_count=95, exported_invite=None, migrated_from_chat_id=None, migrated_from_max_id=None, pinned_msg_id=11328, stickerset=StickerSet(id=351448540947415046, access_hash=3409533151634875646, title='tgcalls', short_name='tgcalls', count=4, hash=-1742156260, archived=False, official=False, masks=False, animated=True, installed_date=None, thumbs=[PhotoPathSize(type='j', bytes=b'\x19\x03\xb3\x02\xdck\x80X\x00EY\x01xEY\x85X\x89g\x81DDKCQ\x81F\x88E\x89H\x82E\x88]\x87b\x81DGDJGRRQuIG\x06\x88V\x99P\xa1\\\x8aL\x9bl\xa3z\x92a\x88\x03W\x8a\x07F\x90\x8b\x99\xb3\xa3\x86\x05\x89\x8c\x96\x91\x99\xa3\x83\x8d\x82\x9eA\xab\x80\x83J\x91J\x92\x81\x83\x86\x84\x88\x87\x85\x85\x89\x94\x8f\x98\x89\x85\x9b\x80\x9fF\x81A\x88H\x86JUQc_\\~\x83S\x8dX\x98e\x8bN\x8d~\xa3G\x06\x90I\xa0L\xb1N\x82\x80\x85\x81\x86\x81\x86KFI\x06\x83I\t\x8eD\x88\x9a\x88\xa1\x80\x90H\xb6\x80\x86\x08\x82\x84\x9d\x8a\xa2\x8f\x8e\x8b\x97\xa1\xa2\xb0\x8c\x90\xa0\x9b\xa9\xad\x8b\x94\x89\xb2E\x86\x08E\x85S\x85Q\x8c\x82\x88\x87\xab\x8a\xaf\x8a\x90\x9f\x84\x9b\xaeF\xb7U\x06\xbbY\x07\xbb'), PhotoSize(type='m', w=100, h=100, size=14136)], thumb_dc_id=2, thumb_version=-761908018), available_min_id=None, folder_id=None, linked_chat_id=1410595867, location=None, slowmode_seconds=None, slowmode_next_send_date=None, stats_dc=None, call=InputGroupCall(id=-6067902465567988211, access_hash=7136850119039885839), ttl_period=None, pending_suggestions=[], groupcall_default_join_as=PeerUser(user_id=942356877))
```

Reply:
> Telegram sets the user default join as the connected user

we manually pass join as arg to joinGroupCallRequest…

Reply:
[**@MarshalX**](https://github.com/MarshalX) commented on [Jul 17, 2021, 1:30 PM GMT+3](https://github.com/MarshalX/tgcalls/pull/102#discussion_r671669334 "2021-07-17T10:30:33Z - Replied by Github Reply Comments"):
> > Telegram sets the user default join as the connected user
> 
> we manually pass join as arg to joinGroupCallRequest…

Ok. So you want to remove `groupcall_default_join_as` ?

Reply:
I was talking that telegram automatically sets `groupcall_default_join_as` as the connected user

Reply:
> telegram automatically sets `groupcall_default_join_as` as the connected user

How is it can be possible? Look at this code https://github.com/MarshalX/tgcalls/blob/46954ce08fcb6ac5e5aefd844ec68aa5bcaea40e/pytgcalls/pytgcalls/mtproto/telethon_bridge.py#L178

we set how to join… not a telegram

Reply:
[**@MarshalX**](https://github.com/MarshalX) commented on [Jul 17, 2021, 1:37 PM GMT+3](https://github.com/MarshalX/tgcalls/pull/102#discussion_r671670019 "2021-07-17T10:37:25Z - Replied by Github Reply Comments"):
> > telegram automatically sets `groupcall_default_join_as` as the connected user
> 
> How is it can be possible? Look at this code
> 
> https://github.com/MarshalX/tgcalls/blob/46954ce08fcb6ac5e5aefd844ec68aa5bcaea40e/pytgcalls/pytgcalls/mtproto/telethon_bridge.py#L178
> 
> we set how to join… not the telegram

I'm not talking about that. I'm talking about the `groupcall_default_join_as` in `full_chat` which is set as the input peer of the account calling the request.

Reply:
> I'm talking about the `groupcall_default_join_as` in `full_chat` which is set as the input peer of the account calling the request.

Emmmm. And? I already know it. I ask you to add using of this field to join if join_as arg not be passed… 

I want to have this join_as logic:
- is arg passed - use this arg
- If arg not passed and default join as exists - use default join as
- Otherwise (if there is no arg and default join as) use self peer to join

Reply:
> 
> 
> > I'm talking about the `groupcall_default_join_as` in `full_chat` which is set as the input peer of the account calling the request.
> 
> Emmmm. And? I already know it. I ask you to add using of this field to join as if join_as arg not be passed…
> 
> I want to have this join_as logic:
> 
>     * is arg passed - use this arg
> 
>     * If arg not passed and default join as exists - use default join as
> 
>     * Otherwise (if there is no arg and default join as) use self peer to join

Wait. I think I understood it wrong. Sorry.

Pushed the changes